### PR TITLE
Happiest Mask No Longer Makes People Sad

### PR DESCRIPTION
**Pull request by Shadowmech88**
The happiest mask now *transmogrifies* the ghosts of dead players into shades, rather than just sticking their client into a shade.
What this means is that the player is not severed from their body, and can die or use the untransmogrify button to turn back into a ghost if they wish to be cloned or defibbed, for example.
Each mind can only be turned into a shade once per death, so you don't get to keep hopping back and forth or coming back as a shade after dying as a shade.
Happiest masks still only affect dead players who are inside their corpses.

Since the mask is now no longer garbage, it has been re-added to Xenoarch finds.

:cl:
 * tweak: Shades created by the happiest mask now have an option to manually turn back into ghosts, or will turn back into ghosts when they die as a shade, meaning that the happiest mask's shade creation no longer prevents cloning/defibbing.
 * rscadd: The happiest mask has been re-added to Xenoarchaeology finds.

### CODE REVIEW

**Comment by SonixApache** [comment]
My issue with adding it back to xenoarch finds:
it still turns people into shades accompanied with a quasi-antag message.
Don't get me wrong, this is definitely better than before, but it might still cause non-antag shitlery.

**Comment by Shadowmech88** [comment]
We could change the latter part to something like "Joyfully carry out [mask man]'s bidding!"

That way they get slapped if they antagonize without being told to and the mask wearer gets slapped if he tells them to antagonize.

**Comment by Gunosaguki** [comment]
How long does it take for the mask to murder the wearer again? I forget

**Comment by Shadowmech88** [comment]
The mask doesn't harm the wearer at all.

**Comment by Gunosaguki** [comment]
Ah, that must be from a while back. I think I remember reading about it on a wiki at one point

**Comment by Shadowmech88** [comment]
It's also possible you misinterpreted a statement about the mask causing *others* to harm you, which is very likely since everyone used to hate the mask and people who wore it, especially near Medbay.

File: code/modules/clothing/masks/mystic.dm [hunk 1]
```diff
@@ -52,20 +52,26 @@
 	flick("happiest_flash", src)
 	canremove = 1
 
+var/list/has_been_shade
```
Comment (PJB3005) [inline code/modules/clothing/masks/mystic.dm]: Initialize this to `= list()`

File: code/modules/clothing/masks/mystic.dm [hunk 2]
```diff
@@ -52,20 +52,26 @@
...
+
 /obj/item/clothing/mask/happy/proc/RaiseShade(var/mob/living/carbon/human/H)
+	if(!has_been_shade)
```
Comment (PJB3005) [inline code/modules/clothing/masks/mystic.dm]: Remove this.

**Review by PJB3005 — DISMISSED** [review]

**Comment by PJB3005** [comment]
Personally I think it should go by the *minds* of the people being turned to shades. Not ckeys.

And make it reset on revival too.

File: code/modules/clothing/masks/mystic.dm (hunk 1 above)
Comment (PJB3005) [reply code/modules/clothing/masks/mystic.dm]: Oh almost forgot: Could you re-run `gen_globals.py` to add this to `globalaccesss.dm`?

Just run `py -3 tools/gen_globals.py vgstation13.dme code/__HELPERS/globalaccess.dm` from the repo root, in a terminal.

Comment (Shadowmech88) [reply code/modules/clothing/masks/mystic.dm]: Should I always initialize global lists in their definition? I thought it was better for performance for them to not be initialized until they are needed.

**Comment by Riekland** [comment]
A literal god among men

File: code/modules/clothing/masks/mystic.dm (hunk 1 above)
Comment (PJB3005) [reply code/modules/clothing/masks/mystic.dm]: It's one empty list. It seriously doesn't matter.

**Comment by nuklearcellphoneg** [comment]
I get the reference

**Comment by Shadowmech88** [comment]
* The mask now catalogs shades based on the mind of their corpse rather than their ckey.
* Being cloned or defibbed will remove you from the list, allowing you to become a shade again if you die again.

**Comment by Shadowmech88** [comment]
Don't merge this just yet.

**Review by PJB3005 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Alright, now this is good to go.

**Pull request merged**